File: src/main/java/net/andrecarbajal/naviMusic/commands/music/SkipCommand.java
package net.andrecarbajal.naviMusic.commands.music;

import com.sedmelluq.discord.lavaplayer.track.AudioTrack;
import net.andrecarbajal.naviMusic.audio.GuildMusicManager;
import net.andrecarbajal.naviMusic.audio.MusicService;
import net.andrecarbajal.naviMusic.commands.SlashCommand;
import net.andrecarbajal.naviMusic.dto.response.Response;
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
import org.springframework.stereotype.Component;

@Component
public class SkipCommand extends SlashCommand {
    private final MusicService musicService;

    public SkipCommand(MusicService musicService) {
        super("skip", "Skip/remove first song from queue");
        this.musicService = musicService;
    }

    @Override
    public void onCommand(SlashCommandInteractionEvent event) {
        if (noVoiceChannelCheck(event)) return;
        musicService.skipTrack(event.getChannel().asTextChannel()).sendReply(event);

        GuildMusicManager musicManager = musicService.getGuildMusicManager(event.getChannel().asTextChannel().getGuild());
        AudioTrack track = musicManager.getPlayer().getPlayingTrack();

        if (track != null) {
            new Response("Now playing: " + track.getInfo().title, Response.Type.OK, false);
        } else {
            new Response("No track playing", Response.Type.ERROR, false);
        }
    }
}

File: src/main/kotlin/net/andrecarbajal/naviMusic/commands/music/SkipCommand.kt
package net.andrecarbajal.naviMusic.commands.music

import net.andrecarbajal.naviMusic.audio.MusicService
import net.andrecarbajal.naviMusic.commands.SlashCommand
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent
import net.dv8tion.jda.api.interactions.commands.OptionType
import net.dv8tion.jda.api.interactions.commands.build.OptionData
import org.springframework.stereotype.Component

@Component
class SkipCommand(private val musicService: MusicService) :
    SlashCommand("skip", "Skip/remove first song from queue", Category.MUSIC) {

    init {
        addOption(OptionData(OptionType.INTEGER, "position", "Position in queue to skip", false))
    }

    override fun onCommand(event: SlashCommandInteractionEvent) {
        if (noVoiceChannelCheck(event)) return

        val position = event.getOption("position")?.asLong?.toInt() ?: 1
        val textChannel = event.channel.asTextChannel()
        musicService.skipTrack(textChannel, position).sendReply(event)
    }
}
